fe/fe-core/src/main/java/com/starrocks/lake/LakeSystemInfoService.java

```diff
@@ -0,0 +1,474 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.lake;
+
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Multimap;
+import com.starrocks.catalog.DiskInfo;
+import com.starrocks.common.DdlException;
+import com.starrocks.common.Pair;
+import com.starrocks.common.Status;
+import com.starrocks.common.UserException;
+import com.starrocks.qe.ShowResultSet;
+import com.starrocks.sql.analyzer.SemanticException;
+import com.starrocks.sql.ast.DropBackendClause;
+import com.starrocks.sql.ast.ModifyBackendAddressClause;
+import com.starrocks.system.Backend;
+import com.starrocks.system.ComputeNode;
+import com.starrocks.system.SystemInfoService;
+import com.starrocks.thrift.TNetworkAddress;
+import com.starrocks.thrift.TStorageMedium;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * SystemInfoService for lake table and warehouse.
+ * Backend is generated from StarOS worker.
+ */
+public class LakeSystemInfoService extends SystemInfoService {
+    private final Map<Long, Long> clusterIdToLastBackendId = Maps.newHashMap();
+    private final StarOSAgent agent;
+
+    public LakeSystemInfoService(StarOSAgent agent) {
+        this.agent = agent;
+    }
+
+    @Override
+    public void addComputeNodes(List<Pair<String, Integer>> hostPortPairs) throws DdlException {
+        throw new SemanticException("not implemented");
+    }
+
+    @Override
+    public boolean isSingleBackendAndComputeNode(long clusterId) {
+        try {
+            return agent.getWorkersByWorkerGroup(Lists.newArrayList(clusterId)).size() == 1;
+        } catch (UserException e) {
+            throw new SemanticException(e.getMessage());
+        }
+    }
+
+    @Override
+    public void addBackends(List<Pair<String, Integer>> hostPortPairs) throws DdlException {
+        throw new SemanticException("not implemented");
+    }
+
+    @Override
+    public void dropBackend(Backend backend) {
+        throw new SemanticException("not implemented");
```

Comment:
how about throwing `UnsupportedOperationException`? `SemanticException` is throwed when doing sql analysis i think.